### DISCUSSION TITLE
WIFI sleeping?

### DISCUSSION THREAD

**Discussion by phillx19090** [rern/rAudio#496] [Bug Reports]
Hi,

Raspi 1 B+
IQAudiO Pi DAC pro
rAudio 20210827

Of necessity I have to use a wifi dongle between laptop -->>router-->>pi. All is well apart from a slight annoyance as the pi goes incommunicado after about an hour. Ping times out but the pi is clearly still alive as it will happily play all day.

Nothing on the router to shut it down, all wifi light switches work etc. 

# ifconfig wlan0
# iwconfig wlan0

wlan0: flags=4163  mtu 1500
        inet 192.168.1.120  netmask 255.255.255.0  broadcast 192.168.1.255
        ether e8:4e:06:57:39:1c  txqueuelen 1000  (Ethernet)

wlan0     IEEE 802.11  ESSID:"ONTITEL_080819"  
          Mode:Managed  Frequency:2.412 GHz  Access Point: 00:E0:4C:AB:28:E8   
          Bit Rate=150 Mb/s   Tx-Power=20 dBm   
          Retry short limit:7   RTS thr=2347 B   Fragment thr:off
          Power Management:off
          Link Quality=40/70  Signal level=-70 dBm  
          Rx invalid nwid:0  Rx invalid crypt:0  Rx invalid frag:0
          Tx excessive retries:0  Invalid misc:27   Missed beacon:0

Any nudge in the right direction greatly appreciated.

Phill

**Comment by phillx19090** [comment]
Appologies. To be clear, the dongle is on the raspi.

**Comment by R101G** [comment]
@phillx19090 
I have no experience of using the B+ but that iwconfig output looks odd, as it includes ethernet information but does not show the dongle name.
The signal level is good and power management off, so I would try disabling anything which might try to grab wlan0. In the settings, click on the cog next to wifi on/off, and disable the AP option. I know there is no onboard wifi, but it might also be worth adding `dtoverlay=disable-wifi` to /boot/config.txt, in case the OS is trying to use something which does not exist.

I assume you have other DHCP devices which stay connected, so there is no chance the router's DHCP lease time is playing up?

**Comment by phillx19090** [comment]
@R101-Github 

Many thanks for the reply and for not saying, "Your very old B+" ;-)

I´`ve added the overlay and will leave it a while. All other wifi switched etc continue to work without problem so I think the router set up is ok. If the config change doesn´t work then it might be the wifi dongle itself. As long as I make some input to the web server the connection stays alive. Way past time to update the pi I think. 

Again, many thanks.

**Comment by R101G** [comment]
@phillx19090 

If it continues to fail, it would be useful to run it with an ethernet connection as well as wifi. Then you can easily check what state it is in after it fails.

**Comment by phillx19090** [comment]
A good idea. I´ll try that if it fails. Currently it´s gone 2 hours and still accessible, which is an hour longer than previously. If I can still access it when I go to bed then I´d call that a result.

Thanks

**Comment by phillx19090** [comment]
Thanks R101, _dtoverlay=disable-wifi_ has done the trick.

**Comment by R101G** [comment]
Glad to hear that, but I have no idea why :)
